gateway-service/src/test/java/org/zowe/apiml/gateway/security/service/zosmf/ZosmfServiceTest.java

```diff
@@ -146,46 +147,119 @@ void newTokenIsCheckedAgainstInvalidateTokensCache() {
     }
 
     @Test
-    void whenNewTokenFoundInInvalidatedTokensCacheItsLTPAtokenIsInvalidatedAgainstZosmf() {
-        Authentication authentication = new UsernamePasswordAuthenticationToken("user", "pass");
+    void givenLoginRequestCredentials_whenAuthenticate_thenAuthenticateWithSuccess() {
+        Authentication authentication = mock(UsernamePasswordAuthenticationToken.class);
+        LoginRequest loginRequest = mock(LoginRequest.class);
         ZosmfService zosmfService = getZosmfServiceSpy();
         doReturn(true).when(zosmfService).loginEndpointExists();
         ZosmfService.AuthenticationResponse responseMock = mock(ZosmfService.AuthenticationResponse.class);
-        doReturn(responseMock).when(zosmfService).issueAuthenticationRequest(any(), any(), any());
-
-        doReturn(true).when(zosmfService).isInvalidated(any());
-        assertThrows(TokenNotValidException.class, () -> zosmfService.authenticate(authentication));
-        verify(zosmfService, times(1)).invalidate(eq(LTPA), any());
-    }
-
-    @Test
-    void testAuthenticateShouldUseInfoEndpointWhenAuthEndpointDoesNotExist() {
-        ZosmfService zosmfService = getZosmfServiceSpy();
+        when(authentication.getCredentials()).thenReturn(loginRequest);
+        doReturn(true).when(zosmfService).loginEndpointExists();
 
-        doReturn(false).when(zosmfService).loginEndpointExists();
-        doReturn("realm").when(zosmfService).getZosmfRealm("http://zosmf:1433/zosmf/info");
         HttpHeaders requestHeaders = getBasicRequestHeaders();
 
         HttpHeaders responseHeaders = new HttpHeaders();
-        responseHeaders.add(HttpHeaders.SET_COOKIE, "LtpaToken2=lt");
+        responseHeaders.add(HttpHeaders.SET_COOKIE, "jwtToken=jt");
         ResponseEntity<String> responseEntity = new ResponseEntity<>("{}", responseHeaders, HttpStatus.OK);
-
+        doReturn(new ZosmfService.AuthenticationResponse(
+            "domain1",
+            Collections.singletonMap(ZosmfService.TokenType.JWT, "jwtToken1"))).when(zosmfService).getAuthenticationResponse(any());
         doReturn(responseEntity).when(restTemplate).exchange(
-            "http://zosmf:1433/zosmf/info",
-            HttpMethod.GET,
+            "http://zosmf:1433/zosmf/services/authenticate",
```

Comment:
Why did you change this?

Reply:
This is a new test, it is just not visible in git diff.